Neos.Flow/Classes/ObjectManagement/DependencyInjection/ProxyClassBuilder.php

```diff
@@ -213,40 +166,71 @@ protected function buildSetInstanceCode(Configuration $objectConfiguration)
 
     /**
      * Renders code to create identifier/type information from related entities in an object.
-     * Used in sleep methods.
      *
-     * @param Configuration $objectConfiguration
-     * @return string
+     * The code is only rendered if one of the following conditions is met:
+     *
+     *   - The class is annotated with Entity
+     *   - The class is annotated with Scope("session")
+     *
+     * Despite the previous condition, the code will not be rendered if the following condition is true:
+     *
+     *   - The class already has a __sleep() method (we assume that the developer wants to take care of serialization themself)
      */
-    protected function buildSerializeRelatedEntitiesCode(Configuration $objectConfiguration)
+    protected function buildSerializeRelatedEntitiesCode(Configuration $objectConfiguration): string
     {
         $className = $objectConfiguration->getClassName();
-        $code = '';
-        if ($this->reflectionService->hasMethod($className, '__sleep') === false) {
-            $transientProperties = $this->reflectionService->getPropertyNamesByAnnotation($className, Flow\Transient::class);
-            $propertyVarTags = [];
-            foreach ($this->reflectionService->getPropertyNamesByTag($className, 'var') as $propertyName) {
-                $varTagValues = $this->reflectionService->getPropertyTagValues($className, $propertyName, 'var');
-                $propertyVarTags[$propertyName] = isset($varTagValues[0]) ? $varTagValues[0] : null;
-            }
-            $code = "        \$this->Flow_Object_PropertiesToSerialize = array();
-        unset(\$this->Flow_Persistence_RelatedEntities);
 
-        \$transientProperties = " . var_export($transientProperties, true) . ";
-        \$propertyVarTags = " . var_export($propertyVarTags, true) . ";
-        \$result = \$this->Flow_serializeRelatedEntities(\$transientProperties, \$propertyVarTags);\n";
+        if ($this->reflectionService->hasMethod($className, '__sleep')) {
+            return '';
         }
-        return $code;
+
+        $scopeAnnotation = $this->reflectionService->getClassAnnotation($className, Flow\Scope::class);
+
+        $doBuildCode =  $this->reflectionService->getClassAnnotation($className, Flow\Entity::class) !== null;
+        $doBuildCode = $doBuildCode || ($scopeAnnotation->value ?? 'prototype') === 'session';
```

Comment:
```suggestion
        $isEntity =  $this->reflectionService->getClassAnnotation($className, Flow\Entity::class) !== null;
        $doBuildCode = $isEntity || ($scopeAnnotation->value ?? 'prototype') === 'session';
```

Reply:
The meaning is actually "should I build the code"? And that variable name is used elsewhere in proxy class building, so I used that here again for consistency.